### PR TITLE
✨ [Story devtools] Add success page for logs

### PR DESCRIPTION
**Pull request by mszylkowski**
image

### CODE REVIEW

File: extensions/amp-story-dev-tools/0.1/amp-story-dev-tools-tab-logs.js
Comment (gmajoulet) [inline extensions/amp-story-dev-tools/0.1/amp-story-dev-tools-tab-logs.js:256]: For both code clarity and performance, please refactor this code to follow the following structure:
1. Build the DOM
2. In a `mutateElement`, append the entire template at once

Comment (mszylkowski) [reply extensions/amp-story-dev-tools/0.1/amp-story-dev-tools-tab-logs.js:256]: That's what I'm doing right now (check the function calls, they create the content and then get attached in a mutate context), but probably the method names could be renamed or commented better

**Comment by mszylkowski** [comment]
Creating the content in the methods and appending in `buildLogsContent_`. 

Also I found that in some cases the custom CSS can interfere with the layout, so we can remove the custom CSS from the story.

**Comment by gmajoulet** [comment]
deferring to @processprocess

File: extensions/amp-story-dev-tools/0.1/amp-story-dev-tools-tab-logs.css
Comment (gmajoulet) [inline extensions/amp-story-dev-tools/0.1/amp-story-dev-tools-tab-logs.css:120]: We can't use GitHub as a free CDN, please include the SVG in the binary

File: extensions/amp-story-dev-tools/0.1/amp-story-dev-tools-tab-logs.js
Comment (processprocess) [inline extensions/amp-story-dev-tools/0.1/amp-story-dev-tools-tab-logs.js:202]: Since this is static it could be a top level html template.
then this method could be removed and a buildSuccessMessageTemplate could be used instead:
```
 const buildSuccessMessageTemplate = (element) => {
  const html = htmlFor(element);
  return html"<div class="i-amphtml-story-dev-tools-logs-success">
    <div class="i-amphtml-story-dev-tools-logs-success-image"></div>
    <h1>Great Job!<br />No issues found</h1>
  </div>";
};
```

The text needs to be in a text element semantics and a11y.
h1 could be used and targeted for styling. Be sure to set `margin: 0` to override the h1 default margin if that seems like a good route.

Going this route would also require 
`logsContainer.appendChild(this.buildLogsTitle_(this.errorList_.length));`
To be in `buildLogsContent_` and removed from `createErrorsList_` which might be okay since that line is repeated in both methods it might make sense in `buildLogsContent_`.

File: extensions/amp-story-dev-tools/0.1/amp-story-dev-tools-tab-logs.js
Comment (processprocess) [inline extensions/amp-story-dev-tools/0.1/amp-story-dev-tools-tab-logs.js:204]: Mentioned in comment above:
This could be moved to `buildLogsContent_` since it's used in `createErrorsList_` and `createSuccessMessage_`.

File: extensions/amp-story-dev-tools/0.1/amp-story-dev-tools-tab-logs.css
Comment (processprocess) [inline extensions/amp-story-dev-tools/0.1/amp-story-dev-tools-tab-logs.css:133]: nit: this could be on the success message node.
It doesn't appear to have any side effects but there are other text children. It would be more readable to have this on the specific node that is intended to have it.

File: extensions/amp-story-dev-tools/0.1/amp-story-dev-tools-tab-logs.css
Comment (processprocess) [inline extensions/amp-story-dev-tools/0.1/amp-story-dev-tools-tab-logs.css:132]: Might not need this.

File: extensions/amp-story-dev-tools/0.1/amp-story-dev-tools-tab-logs.css
Comment (processprocess) [inline extensions/amp-story-dev-tools/0.1/amp-story-dev-tools-tab-logs.css:120]: If this was in the html we could animate the confetti :) It's a lot of template code though.

Or, if it was 2 svgs (backgroundr: computer screen, foreground: confetti) We could layer them on top and transform the foreground layer 🤔. Maybe that's for a different PR though!

Comment (mszylkowski) [reply extensions/amp-story-dev-tools/0.1/amp-story-dev-tools-tab-logs.css:120]: That's an awesome idea, but agree we could have that in another PR. I expect it's just exporting the image separately from Hong and animating the transform/opacity

File: extensions/amp-story-dev-tools/0.1/amp-story-dev-tools-tab-logs.css
Comment (mszylkowski) [reply extensions/amp-story-dev-tools/0.1/amp-story-dev-tools-tab-logs.css:132]: True, this was needed for adding the `\n` on the textContent before, but using templates we can do `<br>`

File: extensions/amp-story-dev-tools/0.1/amp-story-dev-tools-tab-logs.js
Comment (mszylkowski) [reply extensions/amp-story-dev-tools/0.1/amp-story-dev-tools-tab-logs.js:202]: Actually the image had the margin so `h1`'s margin can be used for that. Sending the commit soon!

File: extensions/amp-story-dev-tools/0.1/amp-story-dev-tools-tab-logs.css
Comment (processprocess) [inline extensions/amp-story-dev-tools/0.1/amp-story-dev-tools-tab-logs.css:120]: There is a confetti looking artifact that differs from the mocks.
screen graphic and font weight differs from mock as well. 

Screen Shot 2021-01-14 at 4 08 26 PM

File: extensions/amp-story-dev-tools/0.1/amp-story-dev-tools-tab-logs.css
Comment (processprocess) [inline extensions/amp-story-dev-tools/0.1/amp-story-dev-tools-tab-logs.css:129]: Does this need !important?

File: extensions/amp-story-dev-tools/0.1/amp-story-dev-tools-tab-logs.js
Comment (processprocess) [inline extensions/amp-story-dev-tools/0.1/amp-story-dev-tools-tab-logs.js:159]: As mentioned in chat, not sure if it was related to this PR:
Screen Shot 2021-01-14 at 3 59 16 PM

Comment (mszylkowski) [reply extensions/amp-story-dev-tools/0.1/amp-story-dev-tools-tab-logs.js:159]: Yeah it's from this PR, will fix

File: extensions/amp-story-dev-tools/0.1/amp-story-dev-tools-tab-logs.css
Comment (mszylkowski) [reply extensions/amp-story-dev-tools/0.1/amp-story-dev-tools-tab-logs.css:120]: Font weight is a tricky one, we don't import the semibold weight into the document, but maybe it makes sense to change the import from bold to semibold since bold is not used much

File: extensions/amp-story-dev-tools/0.1/amp-story-dev-tools-tab-logs.js
Comment (processprocess) [inline extensions/amp-story-dev-tools/0.1/amp-story-dev-tools-tab-logs.js:107]: I think this borks it. Maybe it was for debugging?

Looks good to me after reverting this.

Comment (mszylkowski) [reply extensions/amp-story-dev-tools/0.1/amp-story-dev-tools-tab-logs.js:107]: Yes! That's for testing for the meantime. Will revert before merging

**Review by processprocess — APPROVED** [review]
Approving with comment to remove debugging stub in `buildCallback`

**Pull request merged**
